ViAn/Video/videoplayer.cpp

```diff
@@ -204,16 +204,31 @@ bool VideoPlayer::wait_load_read(){
     }
     // Read new frame and notify processing thread
     {
-        std::lock_guard<std::mutex> lk(m_v_sync->lock);
+        m_v_sync->lock.lock();
+        int load_video_at = m_v_sync->frame_index_on_load;
+        m_frame->store(load_video_at);
         load_video();
-        if (!m_capture.read(m_v_sync->frame)) {
+        if (load_video_at != 0) {
+            m_frame->store(load_video_at);
+            m_v_sync->frame_index_on_load = 0;
+
+            // Must unlock before set_frame() since
+            // it calls synced_read() which in turn
+            // uses the same lock
+            m_v_sync->lock.unlock();
+            set_frame();
+            return m_new_frame->load();
+        } else if (m_capture.read(m_v_sync->frame)) {
+            m_new_frame->store(true);
+            m_frame->store(0);
+            m_v_sync->lock.unlock();
+        } else {
             m_is_playing->store(false);
             playback_stopped();
-
+            m_v_sync->lock.unlock();
             return false;
         }
-        m_new_frame->store(true);
     }
-    m_v_sync->con_var.notify_one();
+    m_v_sync->con_var.notify_all();
```

Comment:
The first check calls on `set_frame` which in turn calls `synced_read` which has a notify so that one should be fine. The program will only get to the last else if it can't read the frame. If there is no new frame we don't want to wake the processing thread so it should also be ok. The last part is how it worked before